File: catkin_ws/src/assignment3/src/monitor_node.cpp
#include "ros/ros.h"
#include "assignment3/CollectData.h"

bool collect_data(assignment3::CollectData::Request &req, assignment3::CollectData::Response &res) {
    // Simulate sensor data collection
    res.data.position.x = 1.0;
    res.data.position.y = 2.0;
    res.data.speed = 0.5;
    // Add more sensor data as needed
    return true;
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "monitor_node");
    ros::NodeHandle n;

    ros::ServiceServer service = n.advertiseService("collect_data", collect_data);
    ros::spin();

    return 0;
}

File: src/assignment3/scripts/monitor_node.py
import rospy
from assignment3.srv import CollectData, CollectDataResponse
from assignment3.msg import SensorData

def collect_data(req):
    # Simulate sensor data collection
    data = SensorData()
    data.position.x = 1.0
    data.position.y = 2.0
    data.speed = 0.5
    # Add more sensor data as needed
    return CollectDataResponse(data)

def monitor_server():
    rospy.init_node('monitor_node')
    service = rospy.Service('collect_data', CollectData, collect_data)
    rospy.spin()

if __name__ == "__main__":
    monitor_server()
